collector/__init__.py

```diff
@@ -0,0 +1,590 @@
+
+        if 'FT' in program_id:
+            program_start, program_end = GetFTProgramDates(notes,semester,year, next_year) 
+            # If still undefined, use the values from the previous observation
+            if not program_start:
+
+                proglist = list(self.programs.copy())
+                program_start = self.programs[proglist[-1]]['progstart']
+                program_end = self.programs[proglist[-1]]['progend'] 
+                
+        else:
+
+            
+            beginning_semester_A = Time(year + "-02-01 20:00:00", format='iso')
+            end_semester_A = Time(year + "-08-01 20:00:00", format='iso')
+            beginning_semester_B =  Time(next_year + "-02-01 20:00:00", format='iso')
+            end_semester_B = Time(next_year + "-08-01 20:00:00", format='iso')
+            # This covers 'Q', 'LP' and 'DD' program observations
+            if semester == 'A':
+                program_start = beginning_semester_A
+                # Band 1, non-ToO, programs are 'persistent' for the following semester
+                program_end = beginning_semester_B if band == Band.Band1 else end_semester_A
+
+            else:
+                program_start = end_semester_A
+                program_end = end_semester_B if band == Band.Band1 else beginning_semester_B
+
+
+        # Flexible boundaries - could be type-dependent
+        program_start -= 14. * u.day
+        program_end += 14. * u.day
+
+
+        # Thesis program?
+        thesis = GetThesis(program)
+
+        # ToO status
+        toostat = GetTooStatus(program)
+
+
+        # Used time from Time Accounting Summary (tas) information
+        if tas and program_id in tas:
+            used = tas[program_id]['prgtime']
+        else:
+            used = 0.0 * u.hour
+
+        self.programs[program_id] = {'mode': program_mode, 'band': band, 'progtime': award, 'usedtime': used,
+                                    'thesis': thesis, 'toostatus': toostat,
+                                    'progstart': program_start, 'progend': program_end, 'idx': [], 'groups': []}
+
+        raw_observations, groups = GetObservationInfo(program)
+
+        if not raw_observations:
+            #print('Parser issue to get observation info')
+            return
+
+        for raw_observation, group in zip(raw_observations,groups):
+
+            classes = uniquelist(GetClass(raw_observation))
+            status = GetObsStatus(raw_observation)
+            obs_odb_id = GetObsID(raw_observation)
+
+            if any(obs_class in obsclasses for obs_class in classes) and (status in selection):
+                
+                print('Adding ' + obs_odb_id, end='\r')
+                total_time = GetObsTime(raw_observation)
+                t, r, d = GetTargetCoords(raw_observation) #NOTE: No clue what t, r or d means
+                if t is None:
+                    t = 'None'
+                if r is None:
+                    r = 0.0
+                    d = 0.0
```

Comment:
I think we should change `t`, `r`, and `d` to `target`, `ra`, and `dec` and do something like:
```
if ra is None and dec is None:
   ra, dec = 0.0, 0.0
elif (ra is None) != (dec is None): # this would be a critical error we should handle, i.e. just one of them is none
   raise # some sort of exception
```

Reply:
Checking the function `GetTargetCoords`, `ra` and `dec` are always define at the same time so that last exception can't happen I think. So that would make sense for the code to only check `ra` because in the case `ra` is `None`, `dec` would also be.